tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/tests/TclModelGenerator.xtend

```diff
@@ -0,0 +1,66 @@
+package org.testeditor.tcl.dsl.tests
+
+import javax.inject.Inject
+import org.testeditor.tcl.SpecificationStepImplementation
+import org.testeditor.tcl.TclModel
+import org.testeditor.tcl.TestCase
+import org.testeditor.tcl.impl.TclFactoryImpl
+import org.testeditor.tml.TestStepContext
+import org.testeditor.tml.dsl.tests.TmlModelGenerator
+
+class TclModelGenerator extends TmlModelGenerator {
+	@Inject TclFactoryImpl tclFactory
+
+	def TclModel tclModel() {
+		return tclFactory.createTclModel
+	}
+
+	def TclModel withTestCase(TclModel me, TestCase testCase) {
+		me.test = testCase
+		return me
+	}
+
+	def TestCase testCase(String name) {
+		return tclFactory.createTestCase => [it.name = name]
+	}
+
+	def TclModel withEnvVariable(TclModel me, String ... names) {
+		names.forEach [ varName |
+			me.environmentVariableReferences.add(tclFactory.createEnvironmentVariableReference => [name = varName])
+		]
+		return me
+	}
+
+	def TclModel withPackage(TclModel me, String packageName) {
+		me.^package = packageName
+		return me
+	}
+
+	def TclModel withNamespaceImport(TclModel me, String namespace) {
+		if (me.importSection == null) {
+			me.importSection = xtypeFactory.createXImportSection
+		}
+		me.importSection.importDeclarations += xtypeFactory.createXImportDeclaration => [
+			it.importedNamespace = namespace
+		]
+		return me
+	}
+
+	def TestCase withSpecificationStep(TestCase me, SpecificationStepImplementation step) {
+		me.steps.add(step)
```

Comment:
Calling this method produces more code than the original code.
Instead of `testCase.steps += step` and I would call `testStep.withSpecificationStep(step)`. Don't see the benefit here.